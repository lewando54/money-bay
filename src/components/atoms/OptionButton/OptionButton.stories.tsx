/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { Meta,StoryObj } from '@storybook/react-native'

import OptionButton from './OptionButton'

const meta = {
    component: OptionButton,
    argTypes: {
        variant: { control: 'select', options: ['normal', 'large']},
        text: { control: 'text' },
        isChosen: { control: 'boolean' },
        onClick: { action: 'chosen' }
    }
} satisfies Meta<typeof OptionButton>

export default meta
type Story = StoryObj<typeof OptionButton>

export const Default: Story = {
    args: {
        variant: 'large',
        text: 'USD',
        isChosen: false
    }
}

export const Normal: Story = {
    args: {
        variant: 'normal',
        text: 'USD',
        isChosen: false
    }
}

export const Chosen: Story = {
    args: {
        variant: 'large',
        text: 'USD',
        isChosen: true
    }
}
