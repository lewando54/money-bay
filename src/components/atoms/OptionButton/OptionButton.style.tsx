import { StyleSheet } from 'react-native'

import { MAIN_DARK } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    baseStyle: {
        backgroundColor: 'white',
        alignSelf: 'flex-start',
        paddingHorizontal: 20,
        paddingVertical: 8,
        borderRadius: 10
    },
    chosenStyle: {
        backgroundColor: MAIN_DARK
    },
    baseText: {
        color: MAIN_DARK,
        fontFamily: 'MulishSemiBold',
        fontSize: 16
    },
    chosenText: {
        color: 'white'
    }
})
