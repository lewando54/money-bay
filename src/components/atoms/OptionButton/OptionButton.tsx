import { TouchableOpacity, Text } from 'react-native'
import optionButtonStyle from './OptionButton.style'
import globalThemeStyle from '../../../styling/GlobalTheme.style'

type TOptionButtonVariant = 'large' | 'normal'

interface IOptionButtonProps {
    variant?: TOptionButtonVariant
    text?: string
    isChosen?: boolean
    onClick?: () => void
    testID?: string
}

export default function OptionButton({
    variant='large',
    text='USD',
    isChosen=false,
    onClick,
    testID
}: IOptionButtonProps){
    return (
        <TouchableOpacity
            onPress={onClick} 
            style={[
                optionButtonStyle.baseStyle,
                globalThemeStyle.default_Shadow,
                isChosen && optionButtonStyle.chosenStyle
            ]}
            testID={testID}
        >
            <Text
                style={[
                    optionButtonStyle.baseText,
                    variant == 'normal' && { fontFamily: 'MulishRegular' },
                    isChosen && optionButtonStyle.chosenText
                ]}
            >
                {text}
            </Text>
        </TouchableOpacity>
    )
}
