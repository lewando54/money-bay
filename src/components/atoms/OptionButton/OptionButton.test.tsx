import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import OptionButton from './OptionButton'
describe('OptionButton', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<OptionButton />)
        expect(tree).toMatchSnapshot()
    })

    it('should render correctly when chosen option is enabled', () => {
        const tree = renderer.create(<OptionButton isChosen={true}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render normal variant', () => {
        const tree = renderer.create(<OptionButton variant='normal' />)
        expect(tree).toMatchSnapshot()
    })

    it('should render large variant', () => {
        const tree = renderer.create(<OptionButton variant='large' />)
        expect(tree).toMatchSnapshot()
    })

    it('should call onClick when pressed', () => {
        const onClick = jest.fn()
        const { getByTestId } = render(<OptionButton isChosen={true} onClick={onClick} testID='test'/>)
        fireEvent.press(getByTestId('test'))
        expect(onClick).toBeCalled()
    })
})
