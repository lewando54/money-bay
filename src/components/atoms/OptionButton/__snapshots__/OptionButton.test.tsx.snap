// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 8,
      "shadowColor": "#16196670",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          },
          false,
          false,
        ],
      ]
    }
  >
    USD
  </Text>
</View>
`;

exports[`OptionButton should render correctly when chosen option is enabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "#1B1D4D",
      "borderRadius": 10,
      "elevation": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 8,
      "shadowColor": "#16196670",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          },
          false,
          {
            "color": "white",
          },
        ],
      ]
    }
  >
    USD
  </Text>
</View>
`;

exports[`OptionButton should render large variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 8,
      "shadowColor": "#16196670",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          },
          false,
          false,
        ],
      ]
    }
  >
    USD
  </Text>
</View>
`;

exports[`OptionButton should render normal variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 8,
      "shadowColor": "#16196670",
      "shadowOffset": {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          },
          {
            "fontFamily": "MulishRegular",
          },
          false,
        ],
      ]
    }
  >
    USD
  </Text>
</View>
`;
